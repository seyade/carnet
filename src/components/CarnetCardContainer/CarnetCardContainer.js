import React, { Component } from 'react';
import axios from 'axios';
import Card from '../CarnetCard/CarnetCard';

class CarnetCardContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      carnets: []
    };
  }

  componentDidMount() {
    this.getAllCarnets();
  }

  getAllCarnets() {
    return axios.get('http://localhost:3009/api/carnets')
      .then(response => {
        const carnets = response.data;
        this.setState({ carnets });
      });
  }

  render() {
    return (
      <article className="carnet-card-container row pull-right">
        <h2 className="card-container-title">Card Container</h2>
        {
          this.state.carnets.map((carnet, index) => {
            return <Card carnet={carnet} key={index}/>
          })
        }
      </article>
    );
  }
}

export default CarnetCardContainer;
