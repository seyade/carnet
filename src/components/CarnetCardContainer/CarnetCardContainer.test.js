import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CarnetCardContainer from '../CarnetCardContainer/CarnetCardContainer';

describe('CarnetCardContainer', () => {
  let cardcontainer;

  beforeEach(() => {
    cardcontainer = shallow(<CarnetCardContainer />);
  });

  it('renders CarnetCardContainer as intended', () => {
    const component = renderer.create(<CarnetCardContainer />);
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should display container title', () => {
    expect(cardcontainer.find('.card-container-title').text()).toEqual('Card Container');
  });
});
