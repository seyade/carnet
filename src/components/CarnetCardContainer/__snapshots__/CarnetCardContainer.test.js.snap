exports[`CarnetCardContainer renders CarnetCardContainer as intended 1`] = `
<article
  className="carnet-card-container row pull-right">
  <h2
    className="card-container-title">
    Card Container
  </h2>
</article>
`;
